### ISSUE TITLE
Invalid context 0x0

### ISSUE THREAD

**Issue by Looveh** [react-native-linear-gradient/react-native-linear-gradient#419]
# Bug

App crashes a few seconds after launch when rendering linear gradient.

XCode logs after setting env var `CG_CONTEXT_SHOW_BACKTRACE`:
```
2019-05-31 15:15:25.613416+0200 MyApp[1460:164421] CGContextSetFillColorSpace: invalid context 0x0. Backtrace:
  <<redacted>+90>
   <<redacted>+228>
    <-[BVLinearGradientLayer drawInContext:]+84>
     <-[BVLinearGradientLayer display]+416>
      <<redacted>+208>
       <<redacted>+24>
        <<redacted>+320>
         <<redacted>+578>
          <<redacted>+162>
           <<redacted>+20>
            <<redacted>+282>
             <<redacted>+1358>
              <CFRunLoopRunSpecific+470>
               <CFRunLoopRunInMode+104>
                <GSEventRunModal+80>
                 <UIApplicationMain+150>
                  <main+70>
May 31 15:15:25  MyApp[1460] <Error>: CGContextSetFillColorSpace: invalid context 0x0. Backtrace:
	  <<redacted>+90>
	   <<redacted>+228>
	    <-[BVLinearGradientLayer drawInContext:]+84>
	     <-[BVLinearGradientLayer display]+416>
	      <<redacted>+208>
	       <<redacted>+24>
	        <<redacted>+320>
	         <<redacted>+578>
	          <<redacted>+162>
	           <<redacted>+20>
	            <<redacted>+282>
	             <<redacted>+1358>
	              <CFRunLoopRunSpecific+470>
	               <CFRunLoopRunInMode+104>
	                <GSEventRunModal+80>
	                 <UIApplicationMain+150>
	                  <main+70>
2019-05-31 15:15:25.616632+0200 MyApp[1460:164421] CGContextSetStrokeColorSpace: invalid context 0x0. Backtrace:
  <<redacted>+98>
   <<redacted>+228>
    <-[BVLinearGradientLayer drawInContext:]+84>
     <-[BVLinearGradientLayer display]+416>
      <<redacted>+208>
       <<redacted>+24>
        <<redacted>+320>
         <<redacted>+578>
          <<redacted>+162>
           <<redacted>+20>
            <<redacted>+282>
             <<redacted>+1358>
              <CFRunLoopRunSpecific+470>
               <CFRunLoopRunInMode+104>
                <GSEventRunModal+80>
                 <UIApplicationMain+150>
                  <main+70>
```
## Environment info

React native info output:

```bash
  React Native Environment Info:
    System:
      OS: macOS 10.14.4
      CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
      Memory: 3.14 GB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 11.14.0 - ~/.nvm/versions/node/v11.14.0/bin/node
      Yarn: 1.15.2 - ~/.nvm/versions/node/v11.14.0/bin/yarn
      npm: 6.9.0 - ~/.nvm/versions/node/v11.14.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.2, macOS 10.14, tvOS 12.2, watchOS 5.2
      Android SDK:
        API Levels: 17, 22, 23, 24, 25, 26, 28
        Build Tools: 23.0.1, 23.0.3, 24.0.1, 24.0.2, 25.0.0, 25.0.2, 26.0.1, 26.0.2, 27.0.3, 28.0.1, 28.0.3
        System Images: android-17 | MIPS, android-17 | Google APIs ARM EABI v7a, android-23 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5522156
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.1 => 16.6.1 
      react-native: 0.58.6 => 0.58.6 
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native: 0.59.5
```

Library version: 2.5.4

## Steps To Reproduce

Render a linear gradient. Non-deterministic, once every other app launch it crashes. Removing the rendering of the linear gradient component removes the issue.

**Comment by yurykorzun** [comment]
I see the same issue in my app. It doesn't crash my app, but it breaks the layout. I integrated React Native in a Unity3d project and it started happening after I updated to the latest Unity version. 

I noticed it has nothing to do with the gradient, I see that yoga is failing to determine the correct size of an element.

CGContextSetFillColorSpace: invalid context 0x0. Backtrace:
  <<redacted>+228>
   <<redacted>+296>
    <-[BVLinearGradientLayer drawInContext:]+96>
     <-[BVLinearGradientLayer display]+508>
      <<redacted>+516>
       <<redacted>+580>
        <<redacted>+92>
         <<redacted>+32>
          <<redacted>+412>
           <<redacted>+1436>
            <CFRunLoopRunSpecific+552>
             <GSEventRunModal+100>
              <UIApplicationMain+236>
               <main+212>                <<redacted>+4>

CGContextSetStrokeColorSpace: invalid context 0x0. Backtrace:
  <<redacted>+240>
   <<redacted>+296>
    <-[BVLinearGradientLayer drawInContext:]+96>
     <-[BVLinearGradientLayer display]+508>
      <<redacted>+516>
       <<redacted>+580>
        <<redacted>+92>
         <<redacted>+32>
          <<redacted>+412>
           <<redacted>+1436>
            <CFRunLoopRunSpecific+552>
             <GSEventRunModal+100>
              <UIApplicationMain+236>
               <main+212>                <<redacted>+4>

**Comment by Looveh** [comment]
I believe I found the issue, it was caused by a component that had large dimensions (300 x 100.000 logical pixels) which iOS didn't want to allocate memory for. The issue trickled down from the methods `display` and `drawInContext` in `BVLinearGradientLayer.m`. 

It should be fairly trivial to check for this and log the issue in a comprehensible way, but I'm afraid I'm not familiar with how to best log it.
